homeassistant/components/mpd/media_player.py

```diff
@@ -251,6 +254,30 @@ def media_album_name(self):
         """Return the album of current playing media (Music track only)."""
         return self._currentsong.get("album")
 
+    @property
+    def media_image_hash(self):
+        """Hash value for media image."""
+        file = self._currentsong.get("file")
+        if file:
+            return hashlib.sha256(file.encode("utf-8")).hexdigest()[:16]
+
+        return None
+
+    async def async_get_media_image(self):
+        """Fetch media image of current playing track."""
+        # not all MPD implementations support the `albumart` command
+        if "albumart" not in self._commands:
+            return None, None
+
+        file = self._currentsong.get("file")
+        if file:
+            response = self._client.albumart(file)
```

Comment:
Does this also do I/O? Blocking I/O isn't allowed in coroutines. Use `hass.async_add_executor_job` to schedule sync work in the thread pool.

Reply:
The python-mpd2 should be able to work async, I'll await the albumart response.